x/upgrade/types/proposal.go

```diff
@@ -5,6 +5,9 @@ import (
 )
 
 const (
+	// TODO HV2: ProposalTypeSoftwareUpgrade was removed. What to do with it?
+	//  ProposalTypeCancelSoftwareUpgrade was not even there in heimdall's gov/types/proposal.go. What to do with it?
```

Comment:
`ProposalTypeCancelSoftwareUpgrade` depends on `ProposalTypeSoftwareUpgrade` semantically, so whatever we do with one, we should do with the other.

Reply:
Right, added a note [here](https://github.com/0xPolygon/cosmos-sdk/pull/15) 
This is already filtered out at proposals submission time, but maybe we could even de-register the types going forward.

Reply:
The note added resolves my comment, thanks.